src/components/thumbnail/documents-section.tsx

```diff
@@ -101,6 +111,37 @@ export const DocumentsSection = observer(({ tab, section, stores, scale,
       });
     }
 
+    // listen for changes to the primary and comparison docs
+    useEffect(() => {
+      const disposer = onSnapshot(stores.ui.problemWorkspace, (workspace) => {
+        setAutoExpandDocKeys({
+          primary: workspace.primaryDocumentKey,
+          comparison: workspace.comparisonDocumentKey
+        });
+      });
+      return () => disposer();
+    }, []);
+
+    useEffect(() => {
+      // try to find the primary document and fall back to the comparison document
+      const docKey = sectionDocs.find(doc => autoExpandDocKeys.primary === doc.key)?.key
+                  || sectionDocs.find(doc => autoExpandDocKeys.comparison === doc.key)?.key;
+      if (!isExpanded && docKey && (docKey !== autoExpandToDocKey)) {
+        // autoexpand when not expanded and the section contains a doc key we haven't expanded to yet
+        setAutoExpandToDocKey(docKey);
+        onToggleExpansion(section);
+      } else if (isExpanded && docKey) {
+        // auto scroll to documents when we are expanded and we have a primary or comparison doc in the section
+        const thumbnail = listRef.current?.querySelector(`.list-item[data-thumbnail-key="${docKey}"]`);
+        if (thumbnail && thumbnail.parentElement) {
```

Comment:
This might be more clear if I replaced `thumbnail.parentElement` with `listRef.current` - let me know if you agree and want me to replace it.

Reply:
The `thumbnail` will be `undefined` if `listRef.current` is `undefined`, so the way you have it seems preferable.